xarray/backends/netCDF4_.py

```diff
@@ -291,7 +291,12 @@ def _set_nc_attribute(obj, key, value):
                    'upgrade to netCDF4-python 1.2.4 or greater.')
             raise AttributeError(msg)
     else:
-        obj.setncattr(key, value)
+        try:
+            obj.setncattr(key, value)
+        except AttributeError as e:
+            msg = ('The following exception occurred when attempting to set '
+                   'attribute ({}, {}): "{}"'.format(key, value, e))
```

Comment:
I don't think we need to include the original error message unless you're silencing the traceback (which is possible). Python 3 will include both tracebacks.

I would suggest just writing something like:
````python
        try:
            obj.setncattr(key, value)
        except AttributeError:
            msg = 'failed to write attribute {} with value {!r} to a netCDF file`.format(key, value)
            raise AttributeError(msg)
```

Reply:
Should we add variable name too, if set?

Reply:
Yes, that would probably be helpful!